euphonic/readers/castep.py

```diff
@@ -528,6 +538,7 @@ def read_interpolation_data(
         data_dict['dielectric'] = dielectric*ureg(
             'e**2/(hartree*bohr)').to(dielectric_unit).magnitude
         data_dict['dielectric_unit'] = dielectric_unit
+
     except UnboundLocalError:
```

Comment:
```
from contextlib import suppress

...
with suppress(UnboundLocalError):
    ...
```
Is the more modern way of doing this.

Reply:
I think there are likely better ways of doing this bit without exceptions, anyway. But it seems out of scope for this PR. I could open an Issue to refactor?

Reply:
Nah, just more of a highlight thing. It's not changed in this PR currently, and probably isn't worth doing so until next time you touch the file.